spec/input_validator_spec.rb

```diff
@@ -22,71 +22,46 @@
-            board.move(1)
-            board.move(4)
-            invalid_moves = [2,3,1,4]
-
-            expect(@input_v.get_invalid_moves(board)).to eq(invalid_moves)
-        end
-    end
-
     describe "#valid_move?" do
-        it "should return false if requested move is already in the board's moves array" do
-            board = Board.new
-
-            board.move(2)
-            @input_v.get_invalid_moves(board)
-            expect(@input_v.valid_move?(2)).to eq(false)
+        before(:each) do
+            @board = Board.new
         end
 
-        it "should return false if requested move is already in the board's moves array" do
-            board = Board.new
+        it "should return false if one requested move is already in the board's moves array" do
+            token = "X"
+            @board.move(token, 2)
+            
+            expect(@input_v.valid_move?(@board.moves, 2)).to eq(false)
+        end
 
-            board.move(2)
-            board.move(3)
-            board.move(1)
-            board.move(4)
-            @input_v.get_invalid_moves(board)
-            expect(@input_v.valid_move?(2)).to eq(false)
-            expect(@input_v.valid_move?(1)).to eq(false)
-            expect(@input_v.valid_move?(1)).to eq(false)
-            expect(@input_v.valid_move?(4)).to eq(false)
+        it "should return false if many requested moves are already in the board's moves array" do
+            token = "X"
+            
+            @board.move(token, 3)
+            @board.move(token, 1)
+            @board.move(token, 4)
+          
+            expect(@input_v.valid_move?(@board.moves, 3)).to eq(false)
+            expect(@input_v.valid_move?(@board.moves, 1)).to eq(false)
+            expect(@input_v.valid_move?(@board.moves, 4)).to eq(false)
         end
 
         it "should return true if requested move is not in the board's moves array" do
-            board = Board.new
+            token = "X"
+            @board.move(token, 1)
 
-            board.move(1)
-            @input_v.get_invalid_moves(board)
-            expect(@input_v.valid_move?(3)).to eq(true)
+            expect(@input_v.valid_move?(@board.moves, 3)).to eq(true)
         end
 
         it "should return true if requested moves are not in the board's moves array" do
-            board = Board.new
-
-            board.move(1)
-            board.move(4)
-            board.move(8)
-            @input_v.get_invalid_moves(board)
-            expect(@input_v.valid_move?(2)).to eq(true)
-            expect(@input_v.valid_move?(3)).to eq(true)
-            expect(@input_v.valid_move?(5)).to eq(true)
+            token = "X"
+            @board.move(token, 1)
+            @board.move(token, 4)
+            @board.move(token, 8)
+
+            expect(@input_v.valid_move?(@board.moves, 2)).to eq(true)
+            expect(@input_v.valid_move?(@board.moves, 3)).to eq(true)
+            expect(@input_v.valid_move?(@board.moves, 5)).to eq(true)
```

Comment:
You don't need 3 expectations here. Just one. If you want to test for other scenarios, separate them out into separate tests.